x-pack/platform/plugins/shared/ml/public/application/components/job_selector/job_selector_badge/job_selector_button.tsx

```diff
@@ -0,0 +1,197 @@
+    const viewInMenu: EuiContextMenuPanelItemDescriptor = {
+      name: i18n.translate(
+        'xpack.ml.overview.anomalyDetection.jobContextMenu.viewInSingleMetricViewer',
+        {
+          defaultMessage: 'View in {page}',
+          values: {
+            page: pageToNavigateTo,
+          },
+        }
+      ),
+      icon: 'visLine',
+      onClick: async () => {
+        const mlLocator = share.url.locators.get(ML_APP_LOCATOR);
+        if (!mlLocator) {
+          return;
+        }
+        const singleMetricViewerLink = await mlLocator.getUrl(
+          {
+            page:
+              page === ML_PAGES.SINGLE_METRIC_VIEWER
+                ? ML_PAGES.ANOMALY_EXPLORER
+                : ML_PAGES.SINGLE_METRIC_VIEWER,
+            pageState: {
+              refreshInterval: {
+                display: 'Off',
+                pause: true,
+                value: 0,
+              },
+              jobIds: [jobId],
+              query: {
+                query_string: {
+                  analyze_wildcard: true,
+                  query: '*',
+                },
+              },
+            },
+          },
+          { absolute: true }
+        );
+        navigateToUrl(singleMetricViewerLink);
+
+        closePopover();
+      },
+    };
+    return [
+      {
+        id: 0,
+        items: [
+          {
+            name: i18n.translate('xpack.ml.overview.anomalyDetection.jobContextMenu.jobDetails', {
+              defaultMessage: 'Job details',
+            }),
+            icon: 'editorTable',
+            onClick: () => {
+              setActiveJobId(jobId);
+              setActiveFlyout(FlyoutType.JOB_DETAILS);
+              closePopover();
+            },
+          },
+          {
+            name: i18n.translate('xpack.ml.overview.anomalyDetection.jobContextMenu.removeJob', {
+              defaultMessage: 'Remove from {page}',
+              values: {
+                page:
+                  page === ML_PAGES.ANOMALY_EXPLORER
+                    ? ANOMALY_EXPLORER_TITLE
+                    : SINGLE_METRIC_VIEWER_TITLE,
+              },
+            }),
+            disabled: removeJobIdDisabled,
+            icon: 'minusInCircle',
+            onClick: () => {
+              if (onRemoveJobId) {
+                onRemoveJobId(jobId);
+              }
+              closePopover();
+            },
+          },
+          {
+            isSeparator: true,
+            key: 'sep',
+          } as EuiContextMenuPanelItemDescriptor,
+          viewInMenu,
+          {
+            name: i18n.translate(
+              'xpack.ml.overview.anomalyDetection.jobContextMenu.viewDatafeedCounts',
+              {
+                defaultMessage: 'View datafeed counts',
+              }
+            ),
+            icon: 'visAreaStacked',
+            onClick: () => {
+              setActiveJobId(jobId);
+              setActiveFlyout(FlyoutType.DATAFEED_CHART);
```

Comment:
Is it posible to set the initial time range in the datafeed flyout to the time range currently shown in the view?

Reply:
Updated here [`3bfbf61` (#207141)](https://github.com/elastic/kibana/pull/207141/commits/3bfbf6156a82166f4d78420db0c710a56bbc0f1d)